scripts/release.ts

```diff
@@ -37,6 +37,19 @@ const inquireReleaseType = async (): Promise<ReleaseType> => {
   return (await inquirer.prompt(questions)).releaseType
 }
 
+const inquireCleanBranchOffMain = async (): Promise<boolean> => {
+  const questions: inquirer.QuestionCollection<{ isCleanlyBranched: boolean }> = [
+    {
+      type: 'confirm',
+      name: 'isCleanlyBranched',
+      message: 'Is your branch cleanly branched off origin/main?',
+      default: false, // Defaulting to false to encourage verification
+    },
+  ]
+  const { isCleanlyBranched } = await inquirer.prompt(questions)
+  return isCleanlyBranched
+}
+
```

Comment:
That's a nice safety and all but let's also check for `isMain` first in case we're trying to author hotfix PRs directly off local main or we'll get burned by our own derpness